src/atoms/TextInput/__tests__/TextInput.native.test.js

```diff
@@ -0,0 +1,218 @@
+  it('snapshot testing with label', () => {
+    const { container } = renderWithTheme(<TextInput label="Email" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with filled TextInput', () => {
+    const { container } = renderWithTheme(<TextInput label="Email">{'user@domain.com'}</TextInput>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with prefix', () => {
+    const { container } = renderWithTheme(<TextInput prefix="kg" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with suffix', () => {
+    const { container } = renderWithTheme(<TextInput suffix="kg" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with prefix & suffix', () => {
+    const { container } = renderWithTheme(<TextInput suffix="kg" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with iconLeft ', () => {
+    const { container } = renderWithTheme(<TextInput iconLeft="info" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with iconRight ', () => {
+    const { container } = renderWithTheme(<TextInput iconRight="info" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with iconLeft & iconRight ', () => {
+    const { container } = renderWithTheme(<TextInput iconLeft="info" iconRight="info" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with help text ', () => {
+    const { container } = renderWithTheme(<TextInput helpText="You can enter your email here" />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with character count ', () => {
+    const { container } = renderWithTheme(<TextInput showCharacterCount maxLength={10} />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with help text & character count ', () => {
+    const { container } = renderWithTheme(
+      <TextInput helpText="You can enter your email here" showCharacterCount maxLength={10} />,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with error ', () => {
+    const { container } = renderWithTheme(
+      <TextInput
+        prefix="info"
+        iconRight="info"
+        label="Email"
+        errorText="Some error has occurred"
+      />,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot testing with disabled ', () => {
+    const { container } = renderWithTheme(
+      <TextInput
+        prefix="info"
+        iconRight="info"
+        label="Email"
+        helpText="You can enter your email here"
+        disabled={true}
+      />,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('snapshot after focus', () => {
+    const { getByTestId, container } = renderWithTheme(
+      <TextInput label="Email" testID="snap-test-text-input" />,
+    );
+    const textInput = getByTestId('snap-test-text-input');
```

Comment:
Say no to `getByTestId`, use it as a last resort only. Here you can use something like `getByPlaceholderText`.

Reply:
This is last resort 😢